src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -0,0 +1,378 @@
+                suggestions=[
+                    'bulk catalyst',
+                    'supported catalyst',
+                    'single crystal',
+                    'metal',
+                    'oxide',
+                    '2D catalyst',
+                    'other',
+                    'unkown',
+                ]
+            ),
+        ),
+        links=['https://w3id.org/nfdi4cat/voc4cat_0007014'],
+    )
+
+    form = Quantity(
+        type=str,
+        shape=[],
+        description="""
+          classification of physical form of catalyst
+          """,
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(suggestions=['sieve fraction', 'powder', 'thin film']),
+        ),
+        links=['https://w3id.org/nfdi4cat/voc4cat_0000016'],
+    )
+
+    def populate_results(
+            self, archive: 'EntryArchive', logger) -> None:
+        '''
+        This function copies the catalyst sample information specified in the dict
+         quantities_results_mapping into the results archive of the entry.
+        '''
+
+        sample_obj = self
+
+        add_catalyst(archive)
+        quantities_results_mapping = {
+            'name': 'catalyst_name',
+            'catalyst_type': 'catalyst_type',
+            'preparation_details.preparation_method': 'preparation_method',
+            'surface.surface_area': 'surface_area',
+            'surface.method_surface_area_determination': 'characterization_methods',
+        }
+
+        for ref_attr, catalyst_attr in quantities_results_mapping.items():
+            value = get_nested_attr(sample_obj, ref_attr)
+            if value is not None:
+                try:
+                    setattr(
+                        archive.results.properties.catalytic.catalyst,
+                        catalyst_attr,
+                        value
+                    )
+                except ValueError:
+                    setattr(
+                        archive.results.properties.catalytic.catalyst,
+                        catalyst_attr,
+                        [value],
+                    )
+
+    def add_referencing_methods(
+            self, archive: 'EntryArchive', logger: 'BoundLogger', number=10) -> None:
+        '''
+        This function looks for other entries that reference the sample and checks the
+        entry type and if it finds a ELNXRayDiffration entry it adds 'XRD' to the
+        characterization_methods in result.
+
+        Args:
+            archive (EntryArchive): The archive containing the section that is being
+            normalized.
+            logger('Bound Logger'): A structlog logger.
+            number: specifies the number of referencing entries that are checked,
+            set to 10 by default
+        '''
+        catalyst_sample = self.m_root().metadata.entry_id
+
+        if self.lab_id is None:
+            logger.warn(f'Found no entries with reference: "{catalyst_sample}".')
+            return
```

Comment:
Probably adjust the message and remove the return.

Reply:
yes.